pkg/adapter/serializer/blueprintV2/blueprintV2.go

```diff
@@ -0,0 +1,189 @@
+	convertedDogus := util.Map(blueprint.Dogus, func(dogu domain.Dogu) TargetDogu {
+		newState, err := serializer.ToSerializerTargetState(dogu.TargetState)
+		errorList = append(errorList, err)
+		return TargetDogu{
+			Name:        dogu.GetQualifiedName(),
+			Version:     dogu.Version.Raw,
+			TargetState: newState,
+		}
+	})
+	convertedComponents := util.Map(blueprint.Components, func(component domain.Component) TargetComponent {
+		newState, err := serializer.ToSerializerTargetState(component.TargetState)
+		errorList = append(errorList, err)
+		return TargetComponent{
+			Name:        component.Name,
+			Version:     component.Version.Raw,
+			TargetState: newState,
+		}
+	})
+
+	err := errors.Join(errorList...)
+	if err != nil {
+		return BlueprintV2{}, fmt.Errorf("cannot convert blueprintMask to BlueprintMaskV1 DTO: %w", err)
+	}
+
+	return BlueprintV2{
+		GeneralBlueprint:        serializer.GeneralBlueprint{API: serializer.V2},
+		Dogus:                   convertedDogus,
+		Components:              convertedComponents,
+		RegistryConfig:          RegistryConfig(blueprint.RegistryConfig),
+		RegistryConfigAbsent:    blueprint.RegistryConfigAbsent,
+		RegistryConfigEncrypted: RegistryConfig(blueprint.RegistryConfigEncrypted),
+	}, nil
+}
+
+func convertToBlueprint(blueprint BlueprintV2) (domain.Blueprint, error) {
+	switch blueprint.API {
+	case serializer.V1:
+		return domain.Blueprint{}, fmt.Errorf("blueprint API V1 is deprecated and got removed: " +
+			"packages and cesapp version got removed in favour of components")
+	case serializer.V2:
+	default:
+		return domain.Blueprint{}, fmt.Errorf("unsupported Blueprint API Version: %s", blueprint.API)
+	}
+	convertedDogus, doguErr := convertDogus(blueprint.Dogus)
+	convertedComponents, compErr := convertComponents(blueprint.Components)
+	err := errors.Join(doguErr, compErr)
+	if err != nil {
+		return domain.Blueprint{}, fmt.Errorf("syntax of blueprintV2 is not correct: %w", err)
+	}
+	return domain.Blueprint{
+		Dogus:                   convertedDogus,
+		Components:              convertedComponents,
+		RegistryConfig:          domain.RegistryConfig(blueprint.RegistryConfig),
+		RegistryConfigAbsent:    blueprint.RegistryConfigAbsent,
+		RegistryConfigEncrypted: domain.RegistryConfig(blueprint.RegistryConfigEncrypted),
+	}, nil
+}
+
+func convertDogus(dogus []TargetDogu) ([]domain.Dogu, error) {
+	var convertedDogus []domain.Dogu
+	var errorList []error
+
+	for _, dogu := range dogus {
+		doguNamespace, doguName, err := serializer.SplitDoguName(dogu.Name)
+		if err != nil {
+			errorList = append(errorList, err)
+			continue
+		}
+		newState, err := serializer.ToDomainTargetState(dogu.TargetState)
+		if err != nil {
+			errorList = append(errorList, err)
+			continue
+		}
+		var version core.Version
+		if dogu.Version != "" {
+			version, err = core.ParseVersion(dogu.Version)
+			if err != nil {
+				errorList = append(errorList, fmt.Errorf("could not parse version of TargetDogu: %w", err))
```

Comment:
```suggestion
				errorList = append(errorList, fmt.Errorf("could not parse version of target dogu %q: %w", dogu.Name, err))
```
Someone reading this message will want to know which dogu